src/CommunicationService/CallCommunicationService.java

```diff
@@ -0,0 +1,14 @@
+package CommunicationService;
+
+import channel.ChannelType;
+
+class CallCommunicationService extends CommunicationService{
```

Comment:
You don't need the CallCommunicationService TextCommunicationService etc. Just one implementation f9.